packages/apex-node/src/utils/traceFlags.ts

```diff
@@ -0,0 +1,233 @@
+          throw new Error(
+            nls.localize(
+              'trace_flags_failed_to_find_debug_level',
+              debugLevelName
+            )
+          );
+        }
+      } else {
+        debugLevelId = await this.createDebugLevel(DEFAULT_DEBUG_LEVEL_NAME);
+        if (!debugLevelId) {
+          throw new Error(
+            nls.localize('trace_flags_failed_to_create_debug_level')
+          );
+        }
+      }
+
+      // create a trace flag
+      const expirationDate = this.calculateExpirationDate(new Date());
+      if (!(await this.createTraceFlag(userId, debugLevelId, expirationDate))) {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  private async findDebugLevel(
+    debugLevelName: string
+  ): Promise<string | undefined> {
+    const query = util.format(
+      "SELECT Id FROM DebugLevel WHERE DeveloperName = '%s'",
+      debugLevelName
+    );
+    const result = (await this.connection.tooling.query(query)) as QueryRecord;
+    return result.totalSize && result.totalSize > 0 && result.records
+      ? result.records[0].Id
+      : undefined;
+  }
+
+  private async updateDebugLevel(id: string): Promise<boolean> {
+    const debugLevel = {
+      Id: id,
+      ApexCode: 'FINEST',
+      Visualforce: 'FINER'
+    };
+    const result = (await this.connection.tooling.update(
+      'DebugLevel',
+      debugLevel
+    )) as DataRecordResult;
+    return result.success;
+  }
+
+  private async createDebugLevel(
+    debugLevelName: string
+  ): Promise<string | undefined> {
+    const developerName = debugLevelName;
+    const debugLevel = {
+      developerName,
+      MasterLabel: developerName,
+      ApexCode: 'FINEST',
+      Visualforce: 'FINER'
+    };
+    const result = (await this.connection.tooling.create(
+      'DebugLevel',
+      debugLevel
+    )) as DataRecordResult;
+    return result.success && result.id ? result.id : undefined;
+  }
+
+  private async updateTraceFlag(
+    id: string,
+    expirationDate: Date
+  ): Promise<boolean> {
+    const traceFlag = {
+      Id: id,
+      StartDate: '',
+      ExpirationDate: expirationDate.toUTCString()
+    };
+    const result = (await this.connection.tooling.update(
+      'TraceFlag',
+      traceFlag
+    )) as DataRecordResult;
+    return result.success;
+  }
+
+  private async createTraceFlag(
+    userId: string,
+    debugLevelId: string,
+    expirationDate: Date
+  ): Promise<string | undefined> {
+    const traceFlag = {
+      tracedentityid: userId,
+      logtype: LOG_TYPE,
+      debuglevelid: debugLevelId,
+      StartDate: '',
+      ExpirationDate: expirationDate.toUTCString()
+    };
+
+    const result = (await this.connection.tooling.create(
+      'TraceFlag',
+      traceFlag
+    )) as DataRecordResult;
+    return result.success && result.id ? result.id : undefined;
+  }
+
+  private isValidDateLength(expirationDate: Date): boolean {
+    const currDate = new Date().valueOf();
```

Comment:
```suggestion
    const currDate = Date.now();
```

Reply:
Thanks! I also cleaned this up in the traceFlags tests